### PR TITLE
Add the python package and the config and make the contracts kept

### LINKED ISSUE #611: Add import-linter
To avoid making circular imports it would be good to make some rules and add a linter as per https://github.com/DiamondLightSource/mx-bluesky/pull/568 but with the config in a TOML file.

The rules:
- epics depends only on core
- tango depends only on core
- sim depends only on core
- fastcs depends only on core, epics, tango
- plan_stubs can depend on anything

## Acceptance Criteria
- Anything that breaks the above rules errors pre-commit

### PR DESCRIPTION
**Pull request by stan-dot**
implemented as discussed in the issue description

Fixes #611

### CODE REVIEW

File: .github/workflows/_test.yml [hunk 1]
```diff
@@ -49,6 +49,10 @@ jobs:
         with:
           python-version: ${{ inputs.python-version }}
           pip-install: ".[dev]"
+
+      - name: Run import linter
+        run: lint-imports
+
```
Comment (coretl) [inline .github/workflows/_test.yml:55]: This isn't needed as pre-commit runs the test on all files anyway

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     uses: ./.github/workflows/_tox.yml
     with:
       tox: pre-commit,type-checking
-
+
```
Comment (coretl) [inline .github/workflows/ci.yml:17]: Please remove spurious whitespace

File: pyproject.toml [hunk 3]
```diff
@@ -164,3 +165,41 @@ lint.preview = true # so that preview mode PLC2701 is enabled
...
+type = "forbidden"
+source_modules = "ophyd_async.tango"
+forbidden_modules = [
+    "ophyd_async.fastcs",
+    "ophyd_async.plan_stubs",
+    "ophyd_async.sim",
+    "ophyd_async.epics",
+]
+
+
```
Comment (coretl) [inline pyproject.toml:200]: Please can you also add the rules for sim?

**Review by coretl — CHANGES_REQUESTED** [review]
Thanks! This is almost there...

File: pyproject.toml (hunk 3 above)
Comment (stan-dot) [reply pyproject.toml:200]: ah yes I missed that one

File: .github/workflows/_test.yml (hunk 1 above)
Comment (stan-dot) [reply .github/workflows/_test.yml:55]: ah right

**Review by coretl — APPROVED** [review]

**Pull request merged**